server/routerlicious/packages/protocol-base/src/quorum.ts

```diff
@@ -261,9 +301,9 @@ export class Quorum extends TypedEventEmitter<IQuorumEvents> implements IQuorum
      * Updates the minimum sequence number. If the MSN advances past the sequence number for any proposal without
      * a rejection then it becomes an accepted consensus value.  If the MSN advances past the sequence number
      * that the proposal was accepted, then it becomes a committed consensus value.
-     * Returns true if immediate no-op is required.
+     * Returns a IQuorumUpdateMsnResult object with a list of changes, or undefined if nothing changed.
      */
-    public updateMinimumSequenceNumber(message: ISequencedDocumentMessage): boolean {
+    public updateMinimumSequenceNumber(message: ISequencedDocumentMessage): IQuorumUpdateMsnResult | undefined {
         const value = message.minimumSequenceNumber;
         if (this.minimumSequenceNumber !== undefined) {
             if (value < this.minimumSequenceNumber) {
```

Comment:
I feel the "error" event below is not sufficient - it's catastrophic failure, it should throw (at minimum) and result in container being closed (on client), not sure what's the equivalence on server

Reply:
That said, the caller already updated this.minimumSequenceNumber to be equal to value. So it feels like this code is noop, and ideally we make this check real?